test/broadcasting.jl

```diff
@@ -0,0 +1,364 @@
+
+    df = copy(refdf)
+    df[1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[end-1:end] .= [10, 10]
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:, 1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:, 1] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    df = copy(refdf)
+    dfv = @view df[1:2, 2:end]
+    dfr = df[1, 3:end]
+    @test_throws DimensionMismatch df[1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[1] .= rand(2, 1)
+    @test_throws DimensionMismatch dfr[end-1:end] .= rand(3, 1)
+    @test_throws DimensionMismatch df[:, 1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[:, 1] .= rand(2, 1)
+
+    df = copy(refdf)
+    df[:x1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:x2] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[[:x4, :x5]] .= 10
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:, :x1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:, :x2] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    df = copy(refdf)
+    df[:x1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:x2] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[[:x4, :x5]] .= [10, 10]
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:, :x1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:, :x2] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    df = copy(refdf)
+    dfv = @view df[1:2, 2:end]
+    dfr = df[1, 3:end]
+    @test_throws DimensionMismatch df[:x1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[:x2] .= rand(2, 1)
+    @test_throws DimensionMismatch dfr[[:x4, :x5]] .= rand(3, 1)
+    @test_throws DimensionMismatch df[:, :x1] .= rand(3, 1)
+    @test_throws DimensionMismatch dfv[:, :x2] .= rand(2, 1)
```

Comment:
This block tests single-column matrices, but the code above only tests vectors. Maybe change this block to tests vectors, but put the whole set in a loop, wrapping vectors in a function which is `identity` for the first run, and `reshape(x, :, 1)` for the second run?

Reply:
see the remark above